### DISCUSSION TITLE
Computing the outgoing ray direction through a medium like glass

### DISCUSSION THREAD

**Discussion by asprasan** [mitsuba-renderer/mitsuba3#266] [Q&A]
I have done the following
1) Generate primary camera rays using ```mi.Ray3f``` function by specifying a single camera origin and multiple directions (following tutorial on scripting a renderer)
2) Create a scene using ```mi.Scene``` consisting of **two glass slabs** placed at two different locations on z-axis parallel to x-y plane (following the tutorial on caustics optimization)
3) Using ray_intersect, I have computed the intersection of the primary camera rays with the scene. At the output, I get a ```SurfaceInteraction3f``` instance containing the information like uv coordinates, sh_frame, wi, etc.

**Question:** How do I compute the outgoing ray directions after the primary camera rays have traversed through the glass slab? That is, how do I obtain wo from the bsdf assigned to the object.

**Comment by Speierers** [comment]
You will need to use the `BSDF::sample()` function for this, which will give you a new direction according to the BSDF assigned to the glass slabs.
